library_conventions/symbols.adoc

```diff
@@ -0,0 +1,123 @@
+https://github.com/LibrePCB/librepcb-doc/issues?q=is%3Aissue+label%3A%22Conventions%3A+Symbols%22[on GitHub]!
+====
+
+
+[#libraryconventions-symbols-genericspecific]
+==== Generic vs. Specific
+
+Generic components should have generic symbols. For example a diode (let's
+say _1N4007_) doesn't need its own symbol, a generic diode symbol is fine.
+So you should name it something like "Diode" and use the same symbol also
+for all other standard diodes. Of course every kind of diode (e.g. Zener)
+should have its own symbol because they look different.
+
+On the other side, there are many very specific components, for example
+a microcontroller. Even if it's possible to also use generic symbols for
+them (e.g. "32-Pin IC"), you should create a symbol specific for that part
+instead. This way you can choose a reasonable
+<<libraryconventions-symbols-pinplacement,pin placement>>.
+
+
+[#libraryconventions-symbols-naming]
+==== Naming
+
+Following conventions apply to symbol names:
+
+- Language must be American English (en_US)
+- Title case (e.g. "Capacitor Bipolar" instead of "Capacitor bipolar")
+- Singular names, not plural (e.g. "Diode" instead of "Diodes")
+- If reasonable, start with the generic term (e.g. "Supply GND" instead
+  of "GND Supply") to improve navigation in sorted lists (all supply
+  symbols are listed next to each other)
+
+
+// Image spans across multiple sections
+image::img/symbol_stm32.png[alt="Example Symbol",role="right"]
+
+
+[#libraryconventions-symbols-origin]
+==== Origin
+
+The origin _(0, 0)_ must be in the center of the symbol (not inclusing
+text elements). For non-symmetrical symbols it must be as close as
+possible to the center, but still on the 2.54mm grid.
+
+
+[#libraryconventions-symbols-pinplacement]
+==== Pin Placement
+
+- For integrated circuit symbols (i.e. rectangular outline), generally
+  *don't place pins at the top and bottom edges*, but only on the left and
+  the right. This helps to get clear, easily readable schematics.
+- *Group pins by functionality*, not by physical location of the leads or by
+  datasheet. Always keep the typical application circuit in mind and choose
+  pin locations which help to get clear schematics with only few crossed-over
+  net lines. For example put _GND_ exactly 5.08mm below the _VCC_ pin if it's
+  likely that capacitors need to be connected to them (capacitors have a
+  height of 5.08mm). Or place _D+_ and _D-_ of a USB device right on top of
+  each other (with the default distance of 2.54mm) as they are always used
+  as a pair.
+
+
+[#libraryconventions-symbols-pinnaming]
+==== Pin Naming
+
+If the function of a pin is absolutely clear (e.g. anode/cathode of a diode),
+choose its abbreviated functionality as name (e.g. "A" for anode and "C" for
+cathode). If the functionality is not clear in the symbol (because it's
+defined by the component using that symbol), just use numbers starting with
+"1" at top left and incrementing counterclockwise.
```

Comment:
I think this would be better:

```suggestion
"1" at top left and increment them counterclockwise.
```

Although both versions are still a bit hard to read.